akka-http-core/src/main/scala/akka/http/impl/engine/client/OutgoingConnectionBlueprint.scala

```diff
@@ -352,4 +357,8 @@ private[http] object OutgoingConnectionBlueprint {
       override def preStart(): Unit = getNextMethod()
     }
   }
+
+  class UnexpectedConnectionClosureException(outstandingResponses: Int)
+    extends RuntimeException(s"The http server closed the connection unexpectedly before delivering responses for $outstandingResponses outstanding requests")
```

Comment:
can we add the following too?

```
[...]. This may have happened e.g. because of an connection idle-timeout or abrupt termination of the entity stream
```